### PR TITLE
Refactor youtube handler sync

### PR DESCRIPTION
**Pull request by chime3**

### CODE REVIEW

File: src/providers/google/youtube-favourite.ts [hunk 1]
```diff
@@ -1,17 +1,25 @@
...
+    ConnectionOptionType,
 } from "../../interfaces";
 import { SchemaFavouriteContentType, SchemaFavouriteType, SchemaFavourite } from "../../schemas";
 import { google, youtube_v3 } from "googleapis";
 import { GaxiosResponse } from "gaxios";
+import { ItemsRangeTracker } from "../../helpers/itemsRangeTracker";
 
 const _ = require("lodash");
 
+const MAX_BATCH_SIZE = 50;
```
Comment (tahpot) [inline src/providers/google/youtube-favourite.ts:17]: Can you confirm why 50 was chosen and add a comment.

File: src/providers/google/youtube-favourite.ts [hunk 2]
```diff
@@ -27,143 +35,173 @@ export default class YouTubeFavourite extends GoogleHandler {
...
-    }
+            if (currentRange.startId) {
+                query.pageToken = currentRange.startId;
+            }
+
+            const backfillResponse = await youtube.videos.list(query);
+            const backfillResult = await this.buildResults(
+                backfillResponse,
+                currentRange.endId,
+                _.has(this.config, "metadata.breakTimestamp")
```
Comment (tahpot) [inline src/providers/google/youtube-favourite.ts:119]: Youtube doesn't support time ordering of activities.

Remove support for `breakTimestamp` in the handlers and the unit tests for all youtube.

File: src/providers/google/youtube-favourite.ts [hunk 3]
```diff
@@ -27,143 +35,173 @@ export default class YouTubeFavourite extends GoogleHandler {
...
-                this.emit('log', logEvent)
+                };
+                this.emit('log', logEvent);
+                breakHit = SyncItemsBreak.ID;
                 break;
             }
-    
+
             const snippet = video.snippet;
             const insertedAt = snippet.publishedAt || "Unknown";
```
Comment (tahpot) [inline src/providers/google/youtube-favourite.ts:183]: All date/time fields should be converted to a Date() object and then to `ISOString()` for extra guarantees that we get the correct date/time in the correct format.

**Review by tahpot — CHANGES_REQUESTED** [review]
See comments for changes.

File: src/providers/google/youtube-favourite.ts (hunk 1 above)
Comment (chime3) [reply src/providers/google/youtube-favourite.ts:17]: > Can you confirm why 50 was chosen and add a comment.

The `maxResults` parameter in the Youtube Data API is capped at 50. That's why I set the `MAX_BATCH_SIZE` as 50.

File: src/providers/google/youtube-favourite.ts (hunk 2 above)
Comment (chime3) [reply src/providers/google/youtube-favourite.ts:119]: > Youtube doesn't support time ordering of activities.
> 
> Remove support for `breakTimestamp` in the handlers and the unit tests for all youtube.

Removed

File: src/providers/google/youtube-favourite.ts (hunk 3 above)
Comment (chime3) [reply src/providers/google/youtube-favourite.ts]: > All date/time fields should be converted to a Date() object and then to `ISOString()` for extra guarantees that we get the correct date/time in the correct format.

Done

**Comment by tahpot** [comment]
@todo: Fix summary v description

**Comment by chime3** [comment]
> @todo: Fix summary v description


I removed `summary` after the latest schema change.
Do you want to have both? @tahpot

**Review by tahpot — APPROVED** [review]
Ok to merge.

Just add a comment linking to the docs as to why 50 was chosen.

https://github.com/verida/data-connector-server/pull/93#discussion_r1728184338

**Pull request merged**